### DISCUSSION TITLE
Migrating from klokantech/openmaptiles-server to maptiler/tileserver-gl

### DISCUSSION THREAD

**Discussion by vitorgil** [maptiler/tileserver-gl#1071] [General]
Hi,

I am migrating from the older `openmaptiles-server` container to the newer `tileserver-gl`. The main reason for this is the `--public_url` feature, allowing me to serve several APIs on the same host. This works and I am able to use the feature, the container is behind an nginx proxy that gets https requests and just forwards these in http to the container. Yay!

But now another problem arises that the older server offered 4 styles out of the box (configurable in the start wizard). The new `tileserver-gl` does not have a wizard and has only 2 styles available. I tried just copying the styles from the previous container into the new one, and configuring them in a `config.json` file but that was unsuccessful: the server uses the styles differently somehow, certainly how the `url` of the styles is interpreted. This did not work.

Does anybody have tips on how to deploy `tileserver-gl` with four styles? I need to download everything for offline usage (sprites, fonts, etc). I have been searching and trying for hours without success. Thanks a lot.

Cheers,
Vitor

**Comment by acalcutt** [comment]
What are the 4 styles that get included with the old tilserver wizard? I have only used it once a long time ago and it was before my time working on this project.

Most of the openmaptiles styles are on their github. In "releases" they usually have a complete style with sprites included
https://github.com/openmaptiles?q=style&type=all&language=&sort=

fonts are at
https://github.com/openmaptiles/fonts

**Reply by vitorgil** [reply]
The styles available on `openmaptiles-server` are: dark-matter, klokantech-basic, osm-bright and positron.

Thanks! I will check the links straight away and give it a try.

Also, we are using mbtiles as source and for some of them only a hybrid or satellite style is available, even in `openmaptiles-server`. Do you know the reason for this? In principle the data for any mbtiles file could be presented in any style, what am I missing?

**Reply by acalcutt** [reply]
What I would do is start with this as your base config
https://github.com/maptiler/tileserver-gl/releases/download/v1.3.0/test_data.zip
In that are likely the first two styles you are talking about, klokantech-basic and osm-bright

Then I would download the other two styles from here and put them in your styles folder
https://github.com/openmaptiles/positron-gl-style/releases/tag/v1.8
https://github.com/openmaptiles/dark-matter-gl-style/releases/tag/v1.8
(in fact, you may also want to update osm bright, since there have been some schema updates)
https://github.com/openmaptiles/osm-bright-gl-style/releases

Then I would download noto sans fonts, which are used in osm bright style, and put them in the fonts folder
https://github.com/openmaptiles/fonts/releases/tag/v2.0

Lastly, I would update the config.json to specify the paths to these styles you added, something like
https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/tileserver-gl/configs/config-3.1.1.json#L12-L22

**Reply by acalcutt** [reply]
for the mbtiles question. when tileserver starts with a mbtiles file and no config file, it looks at the id of the mbtiles file to determine if it is labeled "openmaptiles". If it is labeled that, it treats it as a openmaptiles file and adds a openmaptiles data source. when there is an openmaptiles data source, the styles that use that data source are visible.

If the mbtiles file is a raster file (just images), then by default it only shows under 'data'. there are no styles included that would use a raster source by default. you could make a style that uses these raster data sources, but you would have to built that into your style. ([example](https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/tileserver-gl/styles/WDB_OSM/style.json#L57-L59))

If the mbtiles is a vector file, but not labeled 'openmaptiles', it will also only show under data. you could make a style that uses these vector data sources, but you would have to built that into your style.
([example](https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/tileserver-gl/styles/WDB_OSM/style.json#L49-L52))

Once you start with a config.json file, you can really set it up with a lot of style combinations
([example](https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/tileserver-gl/configs/config-3.1.1.json))

**Reply by vitorgil** [reply]
Thanks a lot Andrew. I was already doing what you suggest and indeed it all works when I download gl-styles full packages from releases (I had seen these before, only looked at the repo files).

Do you know what about the hybrid satellite style? It is not available on the gl-styles github. And can this style be used with any mbtiles? I can't find this information.

UPDATE: I am going through you previous answer now.
UPDATE2: Ok, I am not sure your answer is about satellite maps. We use vector mbtiles and then have applications using both raster and vector, so that is a must. I am going to play around with the current setup and use other mbtiles files we have and check what is the behavior.

**Reply by acalcutt** [reply]
I've never found a solution I am happy with for self hosted satellite maps. I thought maptiler used to have a free low res satellite map at https://data.maptiler.com/downloads/planet/ but I'm not seeing it now.

In my own project I made color-relief layers that I add to the styles, like
https://tiles.wifidb.net/data/jaxa_color/#3/0.00/0.00
https://tiles.wifidb.net/data/gebco_color_releif/#4/0.00000/-0.00275
( I offer these at https://techidiots.net/downloads/open-base-map-torrents )

I have also used [google satellite source ](https://github.com/acalcutt/wifidb-tileserver-gl/blob/master/tileserver-gl/styles/WDB_SAT/style.json#L49-L54) in testing, but I've never found a good way to properly attribute, since that is usually built into their api.

**Reply by acalcutt** [reply]
unfortunately I don't have any info about the hybrid satellite style maptiler uses. Once you have a satellite data source, it is not hard to modify an existing style and add that in.

**Reply by acalcutt** [reply]
You could also likely use tiles from https://cloud.maptiler.com/tiles/ , you would just have to use the tilejson url with your specific key in your style. I'm not too familiar with the maptiler side of things, but it should be an option.

![image](https://github.com/maptiler/tileserver-gl/assets/3792408/5d00e1d8-340d-4efe-b342-636861ad07a5)
